Source/WebCore/dom/Document.cpp

```diff
@@ -9787,6 +9789,42 @@ void Document::updateContentRelevancyForScrollIfNeeded(const Element& scrollAnch
     return m_contentVisibilityDocumentState->updateContentRelevancyForScrollIfNeeded(scrollAnchor);
 }
 
+RefPtr<ViewTransition> Document::activeViewTransition() const
+{
+    return m_activeViewTransition;
+}
+
+void Document::setActiveViewTransition(RefPtr<ViewTransition> viewTransition)
+{
+    m_activeViewTransition = WTFMove(viewTransition);
+}
+
+Ref<ViewTransition> Document::startViewTransition(RefPtr<ViewTransitionUpdateCallback> updateCallback)
+{
+    Ref viewTransition = ViewTransition::create(*this, WTFMove(updateCallback));
+
+    if (m_activeViewTransition)
+        m_activeViewTransition->skipViewTransition(Exception { ExceptionCode::AbortError, "Old view transition aborted by new view transition."_s });
+
+    m_activeViewTransition = WTFMove(viewTransition);
+    scheduleRenderingUpdate(RenderingUpdateStep::PerformPendingViewTransitions);
+    return *m_activeViewTransition;
+}
+
+void Document::performPendingViewTransitions()
+{
+    if (!m_activeViewTransition)
+        return;
+    Ref protectedActiveViewTransition { *m_activeViewTransition };
```

Comment:
We don't use this pattern anymore. This should be:
```
Ref activeViewTransition = *m_activeViewTransition;
```

Then use `activeViewTransition` below instead of `m_activeViewTransition`